.github/workflows/benchmark.yml

```diff
@@ -19,32 +19,108 @@ on:
         description: 'Number of job splits'
         required: true
         default: '1'
+      run-tcp-ds-query-benchmark:
+        description: 'Run TPCDSQueryBenchmark: true or false'
+        required: false
+        default: 'false'
 
 jobs:
   matrix-gen:
     name: Generate matrix for job splits
     runs-on: ubuntu-20.04
     outputs:
-      matrix: ${{ steps.set-matrix.outputs.matrix }}
+      class: ${{ steps.set-matrix.outputs.class }}
+      split: ${{ steps.set-matrix.outputs.split }}
+      num-split: ${{ steps.set-matrix.outputs.num-split }}
```

Comment:
I think you won't have to pass it via GitHub outputs. `matrix` is being passed to parse a multi element JSON array but for `class` and `num-split`, we could just directly access it by `${{ steps.set-matrix.outputs.class }}` or  `${{ steps.set-matrix.outputs.num-split }}`.

Reply:
yes. right I think your approach is simpler. removed the matrix dimension